File: 3820-number-of-unique-xor-triplets-ii/number-of-unique-xor-triplets-ii.cpp
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> one(nums.begin(), nums.end());
        unordered_set<int> two;

        for(int i=0; i<n; i++){
            for(auto& num:one) two.insert(nums[i]^num);
        }
        for(int i=0; i<n; i++){
            for(auto& num:two) one.insert(nums[i]^num);
        }

        return one.size();
    }
};

File: 3820-number-of-unique-xor-triplets-ii/number-of-unique-xor-triplets-ii.java
class Solution {
    public int uniqueXorTriplets(int[] nums) {
        if(nums.length == 1) {
            return nums.length;
        }

        HashSet<Integer> s1 = new HashSet<>();
        HashSet<Integer> s2 = new HashSet<>();

        for(int i=0; i<nums.length-1; i++) {
            for(int j=i+1; j<nums.length; j++){
                s1.add(nums[i]^nums[j]);
            }
        }

        for(int am:s1){
            for(int s=0; s<nums.length; s++){
                int res = nums[s]^am;
                s2.add(res);
            }
        }

        return s2.size();
    }
}